tests/test_responsetypes.py

```diff
@@ -79,10 +68,18 @@ def test_from_args(self):
             # headers takes precedence over url
             ({'headers': Headers({'Content-Type': ['text/html; charset=utf-8']}),
               'url': 'http://www.example.com/item/'}, HtmlResponse),
+            ({'body': b'Some plain text data with tabs and null bytes',
+              'url': 'http://www.example.com/item/',
+              'headers': Headers({'Content-Type': ['text/html; charset=utf-8'],
+                                  'X-Content-Type-Options': 'nosniff'})}, TextResponse),
+            ({'body': b'\x03\x02\xdf\xdd\x23', 'url': '://www.example.com/item/',
+              'headers': Headers({'Content-Encoding': 'UTF-8'})}, Response),
+            ({'body': b'Some plain text data with tabs and null bytes'}, TextResponse),
+            ({'body': b'<html><head><title>Hello</title></head>'}, HtmlResponse),
+            ({'body': b'<?xml version="1.0" encoding="utf-8"'}, XmlResponse),
             ({'headers': Headers({'Content-Disposition': ['attachment; filename="data.xml.gz"']}),
               'url': 'http://www.example.com/page/'}, Response),
-
-
+            ({'filename': 'file.txt'}, TextResponse),
         ]
```

Comment:
I believe we need to rethink this test.

Before, it was a test of how specific input resulted in different response classes. Now that xtractmime is responsible for determining the right MIME type (and the tests for that hence belong to the `xtractmime` code base), and since the mapping from MIME type to response is responsibility of a different method (`from_mimetype`) and can hence be tested separately, this test needs to be about verifying that we are passing `xtractmime` the right parameters.

You could use `mock` to compare what the method gets with what `xtractmime`. Mock may not be the easiest to use if you have not used it in the past (or even if you have), but it might be the best approach here

Alternatively, you could figure out scenarios that would have a different outcome if we were failing to pass certain parameters to `xtractmime`, and test those. For example, to verify that `xtractmime` gets the `URL` parameter, you could determine a scenario where http_origin being `True` or `False` affects the outcome of this method, and test both cases, what happens when an HTTP URL is passed and what happens when a non-HTTP URL is passed, verifying that the outcome is different, and hence having proof that the URL parameter is taken into account.

Reply:
I have some additional thoughts.

We should probably keep the the old test scenarios around, as a backward-compatibility test, in a separate test function. In other words, you could leave the existing test function as-is, and rename it to something like `test_from_args_pre_xtractmime`.

It would also be great if you could compare the old and xtractmime-based implementations, and figure out behavior differences that are to be expected, so that we can cover these in the release notes as backward-incompatible changes (hopefully for the better) that people should expect. For example, with the new implementation any MIME type with `+xml` not in `CLASSES` would be an XmlResponse (right?); also, any previous mapping based on filename will stop working. It would be great if you could figure out all those changes in behavior, cover all of them in a test function (e.g. `test_from_args_post_xtractmime`) that checks the current behavior.